polus-feature-subsetting-plugin/src/main.py

```diff
@@ -0,0 +1,258 @@
+    parser.add_argument('--percentile', dest='percentile', type=str,
+                        help='Percentile to remove', required=True)
+    parser.add_argument('--removeDirection', dest='removeDirection', type=str,
+                        help='remove direction above or below percentile', required=True)
+    parser.add_argument('--sectionVar', dest='sectionVar', type=str,
+                        help='variables to divide larger sections', required=False)
+    parser.add_argument('--writeOutput', dest='writeOutput', type=str,
+                        help='write output image collection or not', required=False)
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+    csvDir = args.csvDir
+    if (Path.is_dir(Path(args.csvDir).joinpath('images'))):
+        # switch to images folder if present
+        fpath = str(Path(args.csvDir).joinpath('images').absolute())
+    logger.info('csvDir = {}'.format(csvDir))
+    delay = args.delay
+    delay = 0 if delay==None else int(delay)
+    logger.info('delay = {}'.format(delay))
+    feature = args.feature
+    logger.info('feature = {}'.format(feature))
+    filePattern = args.filePattern
+    logger.info('filePattern = {}'.format(filePattern))
+    groupVar = args.groupVar
+    logger.info('groupVar = {}'.format(groupVar))
+    inpDir = args.inpDir
+    if (Path.is_dir(Path(args.inpDir).joinpath('images'))):
+        # switch to images folder if present
+        fpath = str(Path(args.inpDir).joinpath('images').absolute())
+    logger.info('inpDir = {}'.format(inpDir))
+    percentile = float(args.percentile)
+    logger.info('percentile = {}'.format(percentile))
+    removeDirection = args.removeDirection
+    logger.info('removeDirection = {}'.format(removeDirection))
+    sectionVar = args.sectionVar
+    sectionVar = '' if sectionVar is None else sectionVar
+    logger.info('sectionVar = {}'.format(sectionVar))
+    writeOutput = True if args.writeOutput==None else args.writeOutput == 'true'
+    logger.info('writeOutput = {}'.format(writeOutput))
+    outDir = args.outDir
+    logger.info('outDir = {}'.format(outDir))
+    
+    # create metadata and images folder in outDir
+    if not os.path.isdir(os.path.join(outDir, 'images')):
+        os.mkdir(os.path.join(outDir, 'images'))
+    if not os.path.isdir(os.path.join(outDir, 'metadata_files')):
+        os.mkdir(os.path.join(outDir, 'metadata_files'))
+
+    # Surround with try/finally for proper error catching
+    try:
+
+        # Get all file names in csvDir image collection
+        csvDir_files = [f.name for f in Path(csvDir).iterdir() if f.is_file() and "".join(f.suffixes)=='.csv']
+        
+        # Get all file names in inpDir image collection
+        inpDir_files = [f.name for f in Path(inpDir).iterdir() if f.is_file() and "".join(f.suffixes)=='.ome.tif']
```

Comment:
This is fine for now, but in the future we will be using OME Zarr files. I know this is part of the boilerplate code that is generated from cookiecutter, and it will be changing in the newer version of the template. Is there a way we can make this more file suffix agnostic? Maybe we could do `if '.ome' in f.suffixes`? That way this will still be selective in the data types it will process, but it will only select the types we implement (the new Zarr file format will have the `.ome.zarr` extension).

Reply:
This comment is more of a brainstorming comment than it is a necessity to change for right now. We do not need to make this section of code future proof, so no changes are explicitly required here.